core/primitives/src/network.rs

```diff
@@ -39,38 +56,44 @@ impl From<&PeerId> for Vec<u8> {
 
 impl From<PublicKey> for PeerId {
```

Comment:
Unrelated, but this impl seems semantically wrong. It means that any public key is a PeerId, but that's not true. Some PKs identify things which are not PeerId. We only should have `From` impl in the other direction.

Reply:
@matklad Yeah, also there is no real benefit to doing that. Why not just use `PeerId::new(public_key)`. Having a few ways to do the same thing just makes the code harder to read. If I see somewhere in the code `public_key.into()`. I wouldn't expect it to convert itself to `PeerId`.Those are essentially different types. There is a reason why we don't have `Impl From<u64> for SystemTime`, and other weird usage of `From` like that.

Reply:
I'll do it in another PR.